### PR TITLE
mgr/dashboard: fix ESOCKETTIMEDOUT E2E failure

### PR DESCRIPTION
**Pull request by avanthakkar**
Increasing responseTimeout may help resolve the e2e failures with `cy.visit()/cy.request() failed to load error`, as it's considered the most accepted solution here https://github.com/cypress-io/cypress/issues/7062 too, which happens only in CI jobs. The [top failure jobs in last 3 months](https://dashboard.cypress.io/projects/k7ab29/analytics/top-failures?branches=%5B%5D&browsers=%5B%5D&chartRangeMostCommonErrors=%5B%5D&chartRangeSlowestTests=%5B%5D&chartRangeTopFailures=%5B%5D&committers=%5B%5D&cypressVersions=%5B%5D&flaky=%5B%5D&operatingSystems=%5B%5D&runGroups=%5B%5D&specFiles=%5B%5D&status=%5B%7B%22label%22%3A%22Passed%22%2C%22value%22%3A%22PASSED%22%7D%2C%7B%22label%22%3A%22Failed%22%2C%22value%22%3A%22FAILED%22%7D%5D&tags=%5B%5D&timeInterval=WEEK&timeRange=%7B%22startDate%22%3A%222021-01-30%22%2C%22endDate%22%3A%222021-04-30%22%7D&viewBy=TEST_CASE) have the same error of timeout/failed to load.
Example error message : `cy.request() timed out waiting 30000ms for a response from your server.`
Fixes: https://tracker.ceph.com/issues/49828
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by avanthakkar** [comment]
I think we don't need to make changes in script in order to change the responseTimeout. As when we run `npx cypress run...` it looks for cypress.json file and takes the environment var values from there unless we have changed it in script itself like for `baseURL`. Also no need to change `defaultCommandTimeout` as it's taken as 120s only while running e2e jenkins job.

**Comment by avanthakkar** [comment]
I don't understand why we do this [in afterEach hook ](https://github.com/ceph/ceph/blob/5f12e85d91543db5cd041ef2d840b8f4b98ce6be/src/pybind/mgr/dashboard/frontend/cypress/support/index.ts#L4) ? Shouldn't we instead call something like `cy.logout()`, as for all components testing we login in beforeEach() hook and I think it also keeps up with [best practices](https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks). Thoughts @ceph/dashboard ?

**Comment by alfonsomthd** [comment]
> I don't understand why we do this [in afterEach hook ](https://github.com/ceph/ceph/blob/5f12e85d91543db5cd041ef2d840b8f4b98ce6be/src/pybind/mgr/dashboard/frontend/cypress/support/index.ts#L4) ? Shouldn't we instead call something like `cy.logout()`, as for all components testing we login in beforeEach() hook and I think it also keeps up with [best practices](https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks). Thoughts @ceph/dashboard ?

Try to run the whole suite without the afterEach hook to check if it's not needed anymore.

**Comment by avanthakkar** [comment]
> > I don't understand why we do this [in afterEach hook ](https://github.com/ceph/ceph/blob/5f12e85d91543db5cd041ef2d840b8f4b98ce6be/src/pybind/mgr/dashboard/frontend/cypress/support/index.ts#L4) ? Shouldn't we instead call something like `cy.logout()`, as for all components testing we login in beforeEach() hook and I think it also keeps up with [best practices](https://docs.cypress.io/guides/references/best-practices#Using-after-or-afterEach-hooks). Thoughts @ceph/dashboard ?
> 
> Try to run the whole suite without the afterEach hook to check if it's not needed anymore.

Yes..but I'm sure abt why we redirect to 403 page..I understand atleast why we should redirect (I think it's following best practices in order to clean up the state generated by the current test(s)), but not sure about 403 page in specific..can't we do instead logout?

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Review by aaSharma14 — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/cypress.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   "supportFile": "cypress/support/index.ts",
   "video": false,
   "defaultCommandTimeout": 120000,
+  "responseTimeout": 45000,
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/cypress.json:10]: Are we sure we need to increase the response time from 30 secs to 45 secs? Are we talking about HTTP requests? If that's the case I feel that the issue will be somewhere else :/

Comment (avanthakkar) [reply src/pybind/mgr/dashboard/frontend/cypress.json:10]: > Are we sure we need to increase the response time from 30 secs to 45 secs? Are we talking about HTTP requests? If that's the case I feel that the issue will be somewhere else :/

I followed this https://github.com/cypress-io/cypress/issues/7062#issuecomment-750030997 and I think it's same issue for us. Somehow `cy.request()` is taking some more time in cypress browser env so `cy.visit()` after it is failing to load(also I tried removing that afterEach hook(which isn't consider best practice) but it's somehow broke some tests).

**Review by nizamial09 — APPROVED** [review]

**Review by epuertat — APPROVED** [review]

**Pull request merged**